library/js/limitedcustom_config.js

```diff
@@ -0,0 +1,260 @@
+
+
+                return button;
+            });
+        }
+    }
+
+    class InsertQuestionMark extends CKEDITOR.Plugin {
+        init() {
+            let editor = this.editor;
+            editor.ui.componentFactory.add( 'oeQuestionMark', (locale) => {
+                // The button will be an instance of ButtonView.
+                const button = new CKEDITOR.ButtonView(locale);
+
+                button.set( {
+                    label: 'Insert Question Mark',
+                    tooltip: true,
+                    // AI GENERATION START - generated by ChatGPT 2025-02-24
+                    icon: "<svg \n" +
+                        "    xmlns=\"http://www.w3.org/2000/svg\" \n" +
+                        "    width=\"100\" \n" +
+                        "    height=\"100\" \n" +
+                        "    viewBox=\"0 0 24 24\" \n" +
+                        "    fill=\"none\" \n" +
+                        "    stroke=\"white\" \n" +
+                        "    stroke-width=\"2\" \n" +
+                        "    stroke-linecap=\"round\" \n" +
+                        "    stroke-linejoin=\"round\">\n" +
+                        "    <!-- Circular Border -->\n" +
+                        "    <circle cx=\"12\" cy=\"12\" r=\"10\" />\n" +
+                        "    <!-- Question Mark -->\n" +
+                        "    <path d=\"M12 16h.01\" /> \n" +
+                        "    <path d=\"M12 12c1.5 0 2-1 2-2s-1-2-2-2-2 1-2 2\" />\n" +
+                        "</svg>",
+                    // AI GENERATION END
+                    withText: false
+                } );
+
+                button.on( 'execute', () => {
+                    // Change the model using the model writer.
+                    editor.model.change( (writer) => {
+
+                        // Insert the text at the user's current position.
+                        // CKEDITOR.instances.textarea1.insertText('\r\n');
+                        const textNode = writer.createText('?');
+                        const position = editor.model.document.selection.getFirstPosition();
+                        editor.model.insertContent(textNode, position);
+                    } );
+                } );
+
+
+                return button;
+            });
+        }
+    }
+
+    const {
+        // Alignment,
+        // Autoformat,
+        // AutoImage,
+        // Autosave,
+        // BalloonToolbar,
+        // BlockQuote,
+        // Bold,
+        // Emoji,
+        Essentials,
+        // GeneralHtmlSupport,
+        // Heading,
+        // HorizontalLine,
+        // ImageBlock,
+        // ImageCaption,
+        // ImageInline,
+        // ImageInsertViaUrl,
+        // ImageResize,
+        // ImageStyle,
+        // ImageTextAlternative,
+        // ImageToolbar,
+        // Indent,
+        // IndentBlock,
+        // Italic,
+        // List,
+        // ListProperties,
+        // MediaEmbed,
+        // Mention,
+        // Paragraph,
```

Comment:
uncomment. Takes care of issues and text mode and html mode testing well. I'm going to create some contexts and templates test.

Reply:
Fixed